dist.ts

```diff
@@ -0,0 +1,159 @@
+import { Temporal } from "@js-temporal/polyfill";
+import {
+  BASELINE_LOW_TO_HIGH_DURATION,
+  computeBaseline,
+} from "compute-baseline";
+import { Compat, Feature } from "compute-baseline/browser-compat-data";
+import assert from "node:assert/strict";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import YAML from "yaml";
+
+/**
+ * Update (or create) a dist YAML file from a feature definition YAML file.
+ *
+ * @param {string} fp The path to the human-authored YAML file.
+ */
+function updateDistFile(fp: string): void {
+  const { dir, name: id } = path.parse(fp);
+  const distPath = path.join(dir, `${id}.dist.yml`);
+
+  const stringify = (y: YAML.Document) => y.toString({ lineWidth: 0 });
+
+  const currentDist = fs.existsSync(distPath)
+    ? YAML.parseDocument(fs.readFileSync(distPath, { encoding: "utf-8" }))
+    : undefined;
+  const newDist = toDist(fp);
+
+  currentDist.commentBefore = undefined; // Suppress the generated comment for comparisons
+
+  // Write a dist only if it's new or modified
+  if (!currentDist || stringify(currentDist) !== stringify(newDist)) {
+    newDist.commentBefore = [
+      `Generated from: ${id}.yml`,
+      `Do not edit this file by hand. Edit the source file instead!`,
+      `Last modified: ${Temporal.Now.zonedDateTimeISO("UTC")}`,
+    ]
+      .map((line) => ` ${line}`)
+      .join("\n");
+
+    fs.writeFileSync(distPath, stringify(newDist));
+  }
+}
+
+/**
+ * Generate a dist YAML document from a feature definition YAML file path.
+ *
+ * This passes through most of the contents of a feature definition. If
+ * possible, it fills in `compat_features` from @mdn/browser-compat-data and, if
+ * successful, generates a `status` block.
+ */
+function toDist(fp: string): YAML.Document {
+  const yaml = YAML.parseDocument(fs.readFileSync(fp, { encoding: "utf-8" }));
```

Comment:
If this function looks a bit complicated, it is! This works the way it does to carry forward the order and comments from the authored YAML into the generated YAML.

We could probably simplify this a bit if we wrote a separate formatter to run on _both_ authored and generated files (e.g., to ensure a specific key order, line lengths, etc.)—but that's bit of a separate task and not on the critical path to generating statuses.